### PR TITLE
Copy InstanceID from OpenStackServer to OpenStackMachine

### PR DESCRIPTION
**Pull request by tsuzu**
**What this PR does / why we need it**:
* `OpenStackMachine.Status.InstanceID` is always empty if even the server is running
    * Copy the InstanceID from OpenStackServer to OpenStackMachine as soon as the server gets running

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] Add a step to check status of OpenStackMachine in E2E

/hold

### CODE REVIEW

File: controllers/openstackmachine_controller.go [hunk 1]
```diff
@@ -423,6 +423,7 @@ func (r *OpenStackMachineReconciler) reconcileMachineState(scope *scope.WithLogg
 
 		// Set properties required by CAPI machine controller
 		openStackMachine.Spec.ProviderID = ptr.To(fmt.Sprintf("openstack:///%s", *openStackServer.Status.InstanceID))
+		openStackMachine.Status.InstanceID = openStackServer.Status.InstanceID
```
Comment (EmilienM) [inline controllers/openstackmachine_controller.go:426]: What problem are we trying to solve here? I think we made that on purpose but I'll like you know what your issue is before we revert that change.

Comment (tsuzu) [reply controllers/openstackmachine_controller.go:426]: Hi!
As an operator of CAPI/CAPO, I have a Kubernetes controller that depends on `OpenStackMachine.Stataus.InstanceID`.
And the disappearance of InstanceID will cause a bug.

Are you planning to delete this field in `v1beta1`?
I understand that `InstanceID` can be still retrieved from `ProviderID`, but I hope the breaking change is delivered with the update of API versions like `v1beta2`.

Thank you.

Comment (EmilienM) [reply controllers/openstackmachine_controller.go:426]: I guess since we're v1beta1, we shouldn't have remove that field. I won't block this PR.

**Pull request merged**